include/circt/Dialect/HW/HWStructure.td

```diff
@@ -518,6 +518,40 @@ def InstanceOp : HWInstanceOpBase<"instance", [HWInstanceLike]> {
   let hasVerifier = 1;
 }
 
+def InstanceChoiceOp : HWInstanceOpBase<"instance_choice", []> {
+  let summary = "Represents an instance with a target-specific reference";
+  let description = [{2
+    This represents an instance to a module which is determined based on the
+    target through the ABI. Besides a default implementation, other targets can
+    be associated with a string, which will later determined which reference
+    is chosen.
+
+    For the purposes of analyses and transformations, it is assumed that any of
+    the targets is a possibility.
+
+    Example:
+    ```mlir
+    %b = hw.instance_choice "inst" sym
+        @TargetDefault or
+        @TargetA if "A" or
+        @TargetB if "B"
+        (a: %a: i32) -> (b: i32)
```

Comment:
not sure, but maybe you're missing a colon here
```suggestion
        : (a: %a: i32) -> (b: i32)
```

Reply:
Oh apologies, I misread this as a trailing type, but it is the argument list, so no colon here.